lib/AST/Decl.cpp

```diff
@@ -2892,10 +2892,7 @@ bool ValueDecl::isRecursiveValidation() const {
 Type ValueDecl::getInterfaceType() const {
   auto &ctx = getASTContext();
 
-  // N.B. This assertion exists to catch new broken callers. It can be removed
-  // with the LazyResolver when the time comes.
-  assert(ctx.getLegacyGlobalTypeChecker()
```

Comment:
Do we have to take this out?  It's catching SourceKit regressions right now #28226.